storage/Utils/SystemCmd.h

```diff
@@ -203,6 +214,62 @@ namespace storage
     };
 
 
+    /**
+     * Exception class for SystemCmd. This is used both to really throw
+     * exceptions (if the 'DoThrow' behaviour was set for the SystemCmd) as
+     * well as for just logging the problem (done in both cases, 'DoThrow' and
+     * 'NoThrow'.
+     */
+    class SystemCmdException: public Exception
+    {
+    public:
+	SystemCmdException( SystemCmd * sysCmd, const string & msg ):
```

Comment:
Right, using const SystemCmd * here is a good idea; SystemCmdException only ever uses const methods of the SystemCmd anyway.

A null pointer in that constructor is really not encouraged; it's just the fallback case if that should happen. Throwing an exception in an exception constructor (which would be one alternative way of handling this) might not be such a good idea. ;-)